### PR TITLE
Fix for dangling pointer in logging

### PR DESCRIPTION
**Pull request by Trisfald**
**Description**
This PR addresses the issue [1091](https://github.com/lynckia/licode/issues/1091).
I changed the return value of toLog() from const char* to std::string.

[] It needs and includes Unit Tests

**Changes in Client or Server public APIs**
No changes

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

**Comment by Trisfald** [comment]
I understand your point - with 2 lines code can get messy pretty fast.
I came up with a possible solution, here there's an [example](https://gist.github.com/Trisfald/0fdc1ecc881f7730428a00c4fd0d35e0).  You can run it with an online compiler (I don't know any fiddle equivalent for c++)

It's about passing strings as they are and using a templated helper to extract the c_str. Tell me if this can be a move in the right direction :)

**Comment by lodoyun** [comment]
Yep, that definitely looks like the best solution.
Do you anticipate any significant impact in performance? I don't particularly see any problem but just in case.

**Comment by Trisfald** [comment]
Shouldn't be any performance implication since the bulk of the logic is executed at compile time.

**Comment by Trisfald** [comment]
Got a working solution that doesn't need any change on the codebase. Log macros can now take std::string as inputs.
The nice things is that by further specializing LogElementForwarder, one can log any custom type with '%s'.

**Comment by Trisfald** [comment]
Yes, that is a good idea! LogElementForwarder should always be optimized away by the compiler, but it's better to have smarter code nonetheless

File: erizo/src/erizo/logger.h [hunk 1]
```diff
@@ -69,68 +70,108 @@ log4cxx::LoggerPtr namespace::logger = log4cxx::Logger::getLogger(logName);
...
 
+DEFINE_ELOG_T(ELOG_TRACET, ELOG_TRACE2)
+DEFINE_ELOG_T(ELOG_DEBUGT, ELOG_DEBUG2)
+DEFINE_ELOG_T(ELOG_INFOT, ELOG_INFO2)
+DEFINE_ELOG_T(ELOG_WARNT, ELOG_WARN2)
+DEFINE_ELOG_T(ELOG_ERRORT, ELOG_ERROR2)
+DEFINE_ELOG_T(ELOG_FATALT, ELOG_FATAL2)
 
+// older versions of log4cxx don't support tracing
+#ifdef LOG4CXX_TRACE
```
Comment (jcague) [inline erizo/src/erizo/logger.h]: good catch!

File: erizo/src/erizo/MediaStream.cpp [hunk 2]
```diff
@@ -252,7 +252,7 @@ void MediaStream::read(std::shared_ptr<DataPacket> packet) {
   // PROCESS RTCP
   RtpHeader *head = reinterpret_cast<RtpHeader*> (buf);
   RtcpHeader *chead = reinterpret_cast<RtcpHeader*> (buf);
-  uint32_t recvSSRC;
+  uint32_t recvSSRC = 0;
```
Comment (jcague) [inline erizo/src/erizo/MediaStream.cpp]: :+1:

**Pull request merged**